packages/application/src/connectionlost.ts

```diff
@@ -22,6 +25,45 @@ export const ConnectionLost: IConnectionLost = async function (
       'JupyterLab will continue trying to reconnect.\n' +
       'Check your network connection or Jupyter server configuration.\n'
   );
+  const buttons = [Dialog.okButton({ label: trans.__('Dismiss') })];
+
+  if (!displayConnectionLost) {
+    return;
+  }
+
+  const key = 'server-connection-error';
+  const promise = connectionLostCachePromise[key];
+
+  if (promise) {
+    return promise;
+  } else {
+    try {
+      const errorDialogPromise: any = showDialog({
```

Comment:
New code should not use `any` except for tests. `any` disables type checking and in past lead to numerous user-facing bugs. This will be enforced with https://github.com/jupyterlab/jupyterlab/pull/16219